hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/UnassignProcedure.java

```diff
@@ -231,6 +231,7 @@ protected void finishTransition(final MasterProcedureEnv env, final RegionStateN
     } else {
       // Remove from in-memory states
       am.getRegionStates().deleteRegion(regionInfo);
+      am.getRegionStates().removeRegionFromServer(regionNode.getRegionLocation(), regionNode);
       env.getMasterServices().getServerManager().removeRegion(regionInfo);
```

Comment:
This is my proposal. The main reason for the issue is that we are missing removing read replicas from this cache. Once the RS the replica once was assigned to crashes, the originated SCP will read this cache and find the **read replica** (that does not really exist anywhere else, at this point), and will submit an AP for it. Adding this extra cleanup, allowed your test to succeed.

Reply:
your proposal is nice, will go ahead with your suggestion.